### PR TITLE
DAOS-6567 logging: Disable checkpatch warning for string concatenation.

### PR DESCRIPTION
**Pull request by ashleypittman**
The decision is to not use spaces around DF_ macros, so disable the
test.
Many areas of the code were using spaces, fix these where there
are only one or two instances per file, but do not do wholesale
changes in files where there were multiple instances.
The other files are mostly in CaRT, so we'll update these as other
changes are made in this area.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by SABollinger — APPROVED** [review]

**Review by liuxuezhao — DISMISSED** [review]

**Review by SABollinger — DISMISSED** [review]

**Review by SABollinger — APPROVED** [review]

File: src/include/daos/tests_lib.h [hunk 1]
```diff
@@ -36,26 +36,21 @@
...
-#define assert_success(r) do {					              \
-	int __rc = (r);							      \
-	if (__rc != 0)							      \
-		fail_msg("Not successful!! Error code: " DF_RC, DP_RC(__rc)); \
+#define assert_success(r)						\
+	do {								\
+		int __rc = (r);						\
+		if (__rc != 0)						\
+			fail_msg("Not successful!! Error code: "	\
+				DF_RC, DP_RC(__rc));			\
```
Comment (SABollinger) [inline src/include/daos/tests_lib.h:44]: Still not aligned

File: src/include/daos/tests_lib.h [hunk 2]
```diff
@@ -36,26 +36,21 @@
...
-	int __rc = (r); \
-	if (__rc != errno) \
-		fail_msg("Expected error code \""DF_RC"\" but found " \
-			"\""DF_RC"\"", DP_RC(errno), DP_RC(__rc)); \
+#define assert_rc_equal(rc, expected_rc)				\
+	do {								\
+		if ((rc) == (expected_rc))				\
+			break;						\
+		print_message("assert_rc_equal: %d != %d\n",		\
+			rc, expected_rc);				\
```
Comment (SABollinger) [inline src/include/daos/tests_lib.h:52]: Alignment

File: src/include/daos/tests_lib.h (hunk 1 above)
Comment (ashleypittman) [reply src/include/daos/tests_lib.h:44]: This is a github rendering issue because the code is using tabs.

File: src/include/daos/tests_lib.h [hunk 3]
```diff
@@ -227,7 +222,7 @@ dts_sgl_init_with_strings(d_sg_list_t *sgl, uint32_t count, char *d, ...);
  */
 void
 dts_sgl_init_with_strings_repeat(d_sg_list_t *sgl, uint32_t repeat,
-	uint32_t count, char *d, ...);
+				 uint32_t count, char *d, ...);
```
Comment (SABollinger) [inline src/include/daos/tests_lib.h:225]: Alignment

File: src/placement/tests/jump_map_place_obj.c [hunk 4]
```diff
@@ -88,13 +88,16 @@ gen_oid(daos_obj_id_t *oid, uint64_t lo, uint64_t hi, daos_oclass_id_t cid)
...
-				false), -DER_INVAL); \
+
+#define assert_invalid_param(pl_map, cid)		\
+	do {						\
+		daos_obj_id_t __oid;			\
+		struct pl_obj_layout *__layout = NULL;	\
+		int rc;					\
+		gen_oid(&__oid, 1, UINT64_MAX, cid);	\
+		rc = plt_obj_place(__oid, &__layout,	\
+				pl_map,	false);		\
```
Comment (SABollinger) [inline src/placement/tests/jump_map_place_obj.c:99]: Alignment

File: src/include/daos/tests_lib.h (hunk 1 above)
Comment (ashleypittman) [reply src/include/daos/tests_lib.h:44]: sorry, you're correct.

File: src/include/daos/tests_lib.h (hunk 3 above)
Comment (ashleypittman) [reply src/include/daos/tests_lib.h:225]: This one is github though, I've just checked it.

File: src/placement/tests/jump_map_place_obj.c (hunk 4 above)
Comment (ashleypittman) [reply src/placement/tests/jump_map_place_obj.c:99]: Done.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**